### PR TITLE
DT-524: Change end journey to be /account-details and move test to fl…

### PR DESCRIPTION
**Pull request by petergphillips**
…uentlenium

### CODE REVIEW

File: src/main/java/uk/gov/justice/digital/hmpps/oauth2server/resource/VerifyMobileController.java [hunk 1]
```diff
@@ -28,46 +28,20 @@
...
+                                  final UserService userService,
+                                  @Value("${application.smoketest.enabled}") final boolean smokeTestEnabled) {
         this.verifyMobileService = verifyMobileService;
         this.telemetryClient = telemetryClient;
         this.userService = userService;
         this.smokeTestEnabled = smokeTestEnabled;
     }
 
-    @PostMapping("/change-mobile")
-    public ModelAndView changeMobile(@RequestParam final String mobile, final Principal principal) {
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/digital/hmpps/oauth2server/resource/VerifyMobileController.java]: moved the ChangeMobileController

File: src/main/java/uk/gov/justice/digital/hmpps/oauth2server/security/UserService.kt [hunk 2]
```diff
@@ -44,9 +44,9 @@ class UserService(private val nomisUserService: NomisUserService,
     return StringUtils.isNotEmpty(user.email) && user.isVerified
   }
 
-  fun isSameAsCurrentVerifiedMobile(username: String, mobile: String): Boolean {
+  fun isSameAsCurrentVerifiedMobile(username: String, mobile: String?): Boolean {
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/digital/hmpps/oauth2server/security/UserService.kt]: needed as mobile can be blank / null

File: src/main/java/uk/gov/justice/digital/hmpps/oauth2server/verify/VerifyMobileService.java [hunk 3]
```diff
@@ -161,14 +161,17 @@ public boolean mobileVerified(final String username) {
...
-    @Getter
     public static class VerifyMobileException extends Exception {
         private final String reason;
 
         public VerifyMobileException(final String reason) {
             super(String.format("Verify mobile failed with reason: %s", reason));
             this.reason = reason;
         }
+
+        public String getReason() {
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/digital/hmpps/oauth2server/verify/VerifyMobileService.java]: needed for kotlin

File: src/test/java/uk/gov/justice/digital/hmpps/oauth2server/integration/ChangeMobileSpecification.kt [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+
+    goTo(loginPage).loginAs("AUTH_CHANGE_MOBILE_VERIFIED")
+
+    goTo(changeMobilePage)
+        .updateMobileAs("07700900321", "07700900321")
+
+    verifyMobileAlreadyPage.isAtPage()
+        .continueToAccountDetailsPage()
+
+    accountDetailsPage.isAt()
```
Comment (mikehalmamoj) [inline src/test/java/uk/gov/justice/digital/hmpps/oauth2server/integration/ChangeMobileSpecification.kt]: Is it possible to change this to work the same as goTo? e.g. `isAt(accountDetailsPage)`

**Pull request merged**